pkg/app/api/service/apiservice/service.proto

```diff
@@ -98,6 +98,7 @@ message GetCommandResponse {
 message RegisterEventRequest {
     string name = 1 [(validate.rules).string.min_len = 1];
     string data = 2 [(validate.rules).string.min_len = 1];
+    map<string,string> labels = 3 [(validate.rules).map.keys.string.min_len = 1, (validate.rules).map.values.string.min_len = 1];
```

Comment:
This validates just there is no empty value for a key? I just want to know this fails if no labels set. Sorry I'm lazy I didn't read the documentation for the `validate`.

Reply:
Yes. No labels case is fine. This checks against the empty string of the key, value.

Reply:
In case you want to know the generated code

``` golang
/ Validate checks the field values on RegisterEventRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *RegisterEventRequest) Validate() error {
        if m == nil {
                return nil
        }

        if utf8.RuneCountInString(m.GetName()) < 1 {
                return RegisterEventRequestValidationError{
                        field:  "Name",
                        reason: "value length must be at least 1 runes",
                }
        }

        if utf8.RuneCountInString(m.GetData()) < 1 {
                return RegisterEventRequestValidationError{
                        field:  "Data",
                        reason: "value length must be at least 1 runes",
                }
        }

        for key, val := range m.GetLabels() {
                _ = val

                if utf8.RuneCountInString(key) < 1 {
                        return RegisterEventRequestValidationError{
                                field:  fmt.Sprintf("Labels[%v]", key),
                                reason: "value length must be at least 1 runes",
                        }
                }

                if utf8.RuneCountInString(val) < 1 {
                        return RegisterEventRequestValidationError{
                                field:  fmt.Sprintf("Labels[%v]", key),
                                reason: "value length must be at least 1 runes",
                        }
                }

        }

        return nil
}

```

Reply:
That more makes sense! This generated code tell me how it works completely.